### ISSUE TITLE
False negative pattern irrefutable for literal

### ISSUE THREAD

**Issue by A4-Tacks** [rust-lang/rust-analyzer#22754]
**rust-analyzer version**: rust-analyzer 0.4.2957-standalone

**rustc version**: rustc 1.97.0-nightly (36ba2c771 2026-04-23)

**editor or extension**: Vim

**code snippet to reproduce**:
```rust
fn main() {
    match 0 {
        0 => (),
    }
    match Some(0) {
        Some(0) => (),
    }
}
```

**Expect**: missing match arm ...

**Actual**: no any diagnostics

**Comment by ChayimFriedman2** [comment]
We don't implement exhaustiveness checking for literals (and for other things), I have a branch implementing that.

**Comment by A4-Tacks** [comment]
But enum containing literal patterns are also infected, which is not appropriate

**Comment by ChayimFriedman2** [comment]
We don't lower literal patterns for exhaustiveness at all.
